### PR TITLE
Add testing framework to run plugins in Node

### PR DESCRIPTION
**Pull request by paullinator**
---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1203359209045189

### CODE REVIEW

File: src/util/matchJson.ts [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+  // Compare arrays in order:
+  if (aIsArray) {
+    if (a.length !== b.length) return false
+    for (let i = 0; i < a.length; ++i) {
+      if (!matchJson(a[i], b[i])) return false
+    }
+    return true
+  }
+  // These are both regular objects, so grab the keys:
+  const aKeys = Object.getOwnPropertyNames(a)
```
Comment (swansontec) [inline src/util/matchJson.ts:28]: This is the old corrupted copy from the GUI. Please grab the fresh version from edge-server-tools.

File: test/testpartner.ts [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+} from 'edge-core-js'
+
+import edgeExchangePlugins from '../src'
+import { btcCurrencyInfo } from './fakeBtcInfo'
+import { makeFakePlugin } from './fakeCurrencyPlugin'
+import { ethCurrencyInfo } from './fakeEthInfo'
+
+async function main(): Promise<void> {
+  const allPlugins = {
+    bitcoin: makeFakePlugin(btcCurrencyInfo),
```
Comment (swansontec) [inline test/testpartner.ts:15]: This part is nice, injecting the currencyInfo like this.

**Review by swansontec — CHANGES_REQUESTED** [review]
I would like to see some improvements like using the account snapshot feature, or turning this into a utility function so we can unit-test each of the different plugins, but those can be future projects. This is a fine starting point.

The only real issue is the outdated matchJson file you copied.

File: src/util/matchJson.ts (hunk 1 above)
Comment (paullinator) [reply src/util/matchJson.ts]: Done. Updated w/fixup commit

File: test/testpartner.ts (hunk 2 above)
Comment (paullinator) [reply test/testpartner.ts:15]: Thx

**Comment by paullinator** [comment]
I agree we should use snapshots. I'll update the repo with the snapshot work in my Thorchain PR.

**Review by swansontec — DISMISSED** [review]

**Pull request merged**